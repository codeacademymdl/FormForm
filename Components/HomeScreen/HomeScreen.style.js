import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  main: {
    flex: 1,


    backgroundColor: '#dae5f0',

  },
  icon: {
    width: 40,
    height: 40,
  },
  buttonPos: {
    flex: 1,
    justifyContent: 'center',
    marginTop: '80%',
    flexDirection: 'row',
  },
});
