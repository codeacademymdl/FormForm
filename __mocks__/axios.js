const axios = require('axios');

const get = () => ({
  data: {
    form: 'a',
  },
});

const post = (url, obj) => ({ url, obj });

axios.get = get;
axios.post = post;

export default axios;
