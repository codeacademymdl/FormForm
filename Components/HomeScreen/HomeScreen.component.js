import React, { Component } from 'react';
import {
  View, ScrollView, Image, TouchableOpacity,
} from 'react-native';
import axios from 'axios';
import LandingPageHeader from '../LandingPageHeader/LandingPageHeader.component';
import styles from './HomeScreen.style';
import addIcon from '../../assets/add.png';
import FormBlock from '../FormBlock/FormBlock.component';

export default class HomeScreen extends Component {
  state = {
    formData: [],
  }

  async componentDidMount() {
    const fetchedData = await axios.get('http://localhost:8080/FormData');
    const forms = fetchedData.data.form;
    this.setState({
      formData: forms,
    });
  }

  render() {
    // console.log('inside homescreen', this.props);
    // console.log(this.state.formData);
    return (
      <ScrollView style={styles.main}>
        <LandingPageHeader heading="Awesome Forms" />
        <View style={styles.body}>
          {
            this.state.formData.map(object => <FormBlock key={object.formName} title={object.formName} date={object.createdAt} />)
          }

          <View style={styles.buttonPos}>

            <TouchableOpacity
              underlayColor="blue"
              onPress={() => this.props.navigation.navigate('CreateForm')}
            >
              <Image
                source={addIcon}
                style={styles.icon}
              />
            </TouchableOpacity>

          </View>


        </View>
      </ScrollView>
    );
  }
}
