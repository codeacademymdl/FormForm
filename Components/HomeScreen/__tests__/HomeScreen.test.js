import React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import axios from 'axios';
import HomeScreen from '../HomeScreen.component';


describe('HomeScreen', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<HomeScreen />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('ComponentDidMount', async () => {
    const wrapper = shallow(<HomeScreen />);
    expect(wrapper.instance().state.formData.length === 0).toBe(true);
    axios.get = jest.fn(axios.get);
    await wrapper.instance().componentDidMount();
    // expect(axios.get).toHaveBeenCalled();
    expect(wrapper.instance().state.formData.length > 0).toBe(true);
  });
});
