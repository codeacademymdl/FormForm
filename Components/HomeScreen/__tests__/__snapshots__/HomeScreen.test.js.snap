// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#dae5f0",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4154af",
          "flexDirection": "row",
          "height": 60,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "padding": 10,
          }
        }
      >
        Awesome Forms
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/add.png",
              }
            }
            style={
              Object {
                "height": 40,
                "width": 40,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
